uk.co.compucorp.civicrm.tasksassignments/js/services/contact.js

```diff
@@ -1,101 +1,115 @@
 
-                Contact.get({
-                    'json': {
-                        'id': id,
-                        'return': 'display_name, sort_name, id, contact_id, contact_type, email',
-                        'debug': config.DEBUG
-                }}, function(data){
+                    Contact.get({
+                        'json': {
+                            'id': id,
+                            'return': 'display_name, sort_name, id, contact_id, contact_type, email',
+                            'debug': config.DEBUG
+                        }
+                    }, function (data) {
 
-                    if (UtilsService.errorHandler(data,'Unable to fetch contacts',deferred)) {
-                        return
+                        if (UtilsService.errorHandler(data, 'Unable to fetch contacts', deferred)) {
+                            return
+                        }
+
+                        deferred.resolve(data.values);
+                    }, function () {
+                        deferred.reject('Unable to fetch contact');
+                    });
+
+                    return deferred.promise;
+                },
+                search: function (input, custom_params) {
+
+                    if (params && typeof params !== 'object') {
+                        return null
                     }
 
-                    deferred.resolve(data.values);
-                },function(){
-                    deferred.reject('Unable to fetch contact');
-                });
+                    var deferred = $q.defer();
 
-                return deferred.promise;
-            },
-            search: function(input, params){
+                    var params = {};
 
-                if (params && typeof params !== 'object') {
-                    return null
-                }
+                    angular.extend(params, custom_params);
 
-                var deferred = $q.defer(),
-                    params = params || {};
+                    params.display_name = input;
+                    params.return = 'display_name, contact_id, contact_type, email';
 
-                Contact.get({
-                    action: 'getlist',
-                    json: {
-                        input: input,
-                        params: params,
-                        debug: config.DEBUG
-                }}, function(data){
+                    Contact.get({
+                        action: 'get',
+                        json: params
+                    }, function (data) {
+                        if (UtilsService.errorHandler(data, 'Unable to fetch contact list', deferred)) {
+                            return
+                        }
 
-                    if (UtilsService.errorHandler(data,'Unable to fetch contact list',deferred)) {
-                        return
-                    }
+                        var contacts = [];
+                        
+                        for (var i in data.values) {
```

Comment:
Not an issue, but I think this could be more tersely rewritten as
```js
deferred.resolve(data.values.map(function (contact) {
    return {
        label: contact.display_name,
        description: [contact.email],
        id: contact.contact_id,
        icon_class: contact.contact_type
    };
}));